cpp/tests/io/parquet_reader_test.cpp

```diff
@@ -2724,3 +2724,37 @@ TYPED_TEST(ParquetReaderPredicatePushdownTest, FilterTyped)
   EXPECT_EQ(result_table.num_columns(), expected->num_columns());
   CUDF_TEST_EXPECT_TABLES_EQUAL(expected->view(), result_table);
 }
+
+TEST_F(ParquetReaderTest, ListsWideTable)
+{
```

Comment:
This gtest roughly takes 15s to run on my machine and may fail on smaller GPUs. Should we keep it even?

Reply:
>  Should we keep it even?

Probably yes, unless the test is no longer needed with the changes in this PR.

Reply:
If there is no way to have a smaller test, long CI is still better than buggy code.

Reply:
10-12s is not too bad. I would be interested on how long it takes to run `compute-sanitizer` on this test.
```
compute-sanitizer gtests/PARQUET_TEST --gtest_filter=ParquetReaderTest.ListsWideTable --rmm_mode=cuda
```

Reply:
Takes roughly `145044 ms` on the same machine without using `--rmm_mode=cuda`. With it, more than 30 mins at least.

Reply:
Ok, that is not good. Currently the memchecks take about 1.5 hours so an extra 30 minutes would be a significant hit.
Generally we try not to do large tests like. Is there some other way to test this? If not we may want to disable it.